File: src/types/api/dataset.ts
export type DatasetSummary = {
  id: string;
  name: string;
  dataset_type: string;
  sample_count: number;
  average_tokens: number;
  updated_at: string;
};

export type DatasetImportInput = {
  name: string;
  dataset_type: string;
  format: string;
  file_name: string;
  content_base64: string;
};

export type DatasetAppendInput = {
  dataset_id: string;
  format: string;
  file_name: string;
  content_base64: string;
};

export type DatasetUpdateInput = {
  id: string;
  name: string;
  dataset_type: string;
};

export type DatasetSamplePreview = {
  id: string;
  sample_index: number;
  prompt: string;
  prompt_preview: string;
  estimated_tokens: number;
};

export type DatasetSamplePageInput = {
  dataset_id: string;
  page: number;
  page_size: number;
  keyword?: string | null;
};

export type DatasetSamplePage = {
  items: DatasetSamplePreview[];
  total: number;
  page: number;
  page_size: number;
};

export type DatasetSampleCreateInput = {
  dataset_id: string;
  prompt: string;
};

export type DatasetSampleUpdateInput = {
  sample_id: string;
  prompt: string;
};

export type DatasetSampleBatchDeleteInput = {
  dataset_id: string;
  sample_ids: string[];
};

export type DatasetExportInput = {
  dataset_id: string;
  format: "jsonl" | "csv" | "txt" | string;
};

export type DatasetExportResult = {
  dataset_id: string;
  format: string;
  file_name: string;
  file_path: string;
  mime_type: string;
  sample_count: number;
  message: string;
};

export type DatasetValidationIssue = {
  kind: string;
  label: string;
  count: number;
  sample_indexes: number[];
};

export type DatasetValidationResult = {
  dataset_id: string;
  status: "passed" | "warning" | "failed" | string;
  checked_at: string;
  sample_count: number;
  issues: DatasetValidationIssue[];
  recommendations: string[];
};

File: src-tauri/src/models/dataset.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize)]
pub struct DatasetSummary {
    pub id: String,
    pub name: String,
    pub dataset_type: String,
    pub sample_count: i64,
    pub average_tokens: i64,
    pub updated_at: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetImportInput {
    pub name: String,
    pub dataset_type: String,
    pub format: String,
    pub file_name: String,
    pub content_base64: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetAppendInput {
    pub dataset_id: String,
    pub format: String,
    pub file_name: String,
    pub content_base64: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetUpdateInput {
    pub id: String,
    pub name: String,
    pub dataset_type: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct DatasetSamplePreview {
    pub id: String,
    pub sample_index: i64,
    pub prompt: String,
    pub prompt_preview: String,
    pub estimated_tokens: i64,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetSamplePageInput {
    pub dataset_id: String,
    pub page: i64,
    pub page_size: i64,
    pub keyword: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct DatasetSamplePage {
    pub items: Vec<DatasetSamplePreview>,
    pub total: i64,
    pub page: i64,
    pub page_size: i64,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetSampleCreateInput {
    pub dataset_id: String,
    pub prompt: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetSampleUpdateInput {
    pub sample_id: String,
    pub prompt: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetSampleBatchDeleteInput {
    pub dataset_id: String,
    pub sample_ids: Vec<String>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct DatasetExportInput {
    pub dataset_id: String,
    pub format: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct DatasetExportResult {
    pub dataset_id: String,
    pub format: String,
    pub file_name: String,
    pub file_path: String,
    pub mime_type: String,
    pub sample_count: i64,
    pub message: String,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct DatasetValidationIssue {
    pub kind: String,
    pub label: String,
    pub count: i64,
    pub sample_indexes: Vec<i64>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct DatasetValidationResult {
    pub dataset_id: String,
    pub status: String,
    pub checked_at: String,
    pub sample_count: i64,
    pub issues: Vec<DatasetValidationIssue>,
    pub recommendations: Vec<String>,
}

#[derive(Debug, Clone)]
pub struct DatasetSample {
    pub id: String,
    pub dataset_id: String,
    pub sample_index: i64,
    pub prompt: String,
    pub estimated_tokens: i64,
}
